core/src/test/java/org/apache/calcite/sql/parser/SqlParserTest.java

```diff
@@ -7152,14 +7151,13 @@ public void subTestIntervalSecondFailsValidation() {
     assumeTrue("don't run this test for sub-classes", isNotSubclass());
     // inUrl = "file:/home/x/calcite/core/target/test-classes/hsqldb-model.json"
     String path = "hsqldb-model.json";
-    final URL inUrl = SqlParserTest.class.getResource("/" + path);
-    // URL will convert spaces to %20, undo that
-    String x = URLDecoder.decode(inUrl.getFile(), "UTF-8");
-    assert x.endsWith(path);
-    x = x.substring(0, x.length() - path.length());
-    assert x.endsWith("core/target/test-classes/");
-    x = x.substring(0, x.length() - "core/target/test-classes/".length());
-    final File base = new File(x);
+    File hsqlDbModel = Sources.of(SqlParserTest.class.getResource("/" + path)).file();
+    assert hsqlDbModel.getAbsolutePath().endsWith("core/target/test-classes/hsqldb-model.json")
```

Comment:
it fails on Windows because of separator
something like this works
```
    assert hsqlDbModel.getAbsolutePath().endsWith(
        Paths.get("core", "target", "test-classes", "hsqldb-model.json").toString())
            : hsqlDbModel.getAbsolutePath()
            + " should end with core/target/test-classes/hsqldb-model.json";
```